api/v1/helpers/nouns/generatorTemplates.js

```diff
@@ -36,5 +36,7 @@ module.exports = {
   fieldScopeMap: {
     user: 'user',
   },
+  uniqueFieldWithoutLimit: 'name',
```

Comment:
I don't think this field name makes sense - the "name" fileld isn't actually unique

Reply:
We don't really need a new field for this, all we need to do is specify that this model shouldn't treat name as a unique field by default, whereas all the others do.
It might make more sense to just set the existing "nameFinder" to an empty string, or false/null.
Or change the nameFinder behavior so it always has to be specified.

Reply:
The first approach was as you said... but...

Setting **nameFinder** and using **doFind** (/generatorTemplates/?name=template1): it works fine because it will not go into the limit and when we have same name with more than one version, they will be retrieved.

The problem is when we execute doGet which will look for key (key can be **ID**, **name** or  **nameFinder**). 

**Getting by id (doGet)**

**1)**
/generatorTemplates/**template1** and **nameFinder** = true (for instance as you said).

findBykey, util.js
`opts.where[props.nameFinder || 'name'] = keyClause`

Query will be `where true = template1` and the will not retrieve by name.

**2)**
/generatorTemplates/123-456-789 and **nameFinder** = true.
We will look by ID first and the method **findByIdThenName** removes the where clause retrieving by ID.

Actually for GeneratorTemplate using this approach solves even the story to remove GET by name, however, we are using **nameFinder** to generate a wrong SQL.

I have added this new flag to avoid any side effect related to this concept for nameFinder/uniqueField and 'name' as default option.

And _"the "name" fileld isn't actually unique"_ - I know, but it's default if there is no nameFinder.

Sorry this long comment, but I think now that just satting nameFinder as true will solve as we just look by KEY using ID or name. There are no side effect... hope we can remember why where are executing true = 'templateName' in the future.

Reply:
@iamigo @jgraff2 do you agree setting nameFinder and generating a wrong SQL when searching for generatorTemplate/${name}?